Load bus data from Github

hawkrives opened the pull request:
Closes https://github.com/StoDevX/AAO-React-Native/issues/1540

- Loads data from Github for both the bus routes and the "other modes" list
- Allows the user to "pull to refresh" the bus route (but it really just updates the time)
- [more of a cleanup than anything] Passes data to BusMap instead of letting it load its own data

hawkrives committed:
pull network error reporting into another file

hawkrives committed:
load the Other Modes view's data from github

hawkrives committed:
update BusView and BusMap to load data from the network

BusMap actually now accepts data from a prop, instead of loading anything
on its own.

hawkrives committed:
allow "refreshing" the busview, which triggers a time update

hawkrives committed:
run prettier

hawkrives committed:
add initial loading state to busview

drewvolz commented on source/views/transportation/bus/bus-view.js:
```diff
@@ -1,66 +1,117 @@
+      return defaultData
+    })
+
+    // if (process.env.NODE_ENV === 'development') {
```
@hawkrives

hawkrives replied:
oops

hawkrives committed:
always load local data when developing

drewvolz reviewed (dismissed):
This PR suffers from the same issue as #1534. The list simply isn't updating any changes visually from the interval refreshing.

rye commented on source/views/transportation/bus/bus-view.js:
```diff
@@ -1,66 +1,117 @@
     this.setState(() => ({now: moment.tz(TIMEZONE)}))
   }
 
+  refreshTime = async () => {
```
This function smells a lot like `fetchData`, can any common behavior be extracted out in a convenient way?

hawkrives replied:
I haven't come up with anything.

rye commented on source/views/transportation/bus/bus-view.js:
```diff
@@ -1,66 +1,117 @@
-  static defaultProps = {
-    busLines: defaultBusLines,
-  }
+const GITHUB_URL = 'https://stodevx.github.io/AAO-React-Native/bus-times.json'
```
Can the repo URL get turned into a constant somewhere that this is built off of?  (This is out-of-scope for this PR but an idea nonetheless)

hawkrives replied:
I keep pushing that back to when I want to centralize all data fetching in the redux store… this is, however, unlikely to change, and is easy to grep for.

rye commented on source/views/transportation/other-modes.js:
```diff
@@ -25,6 +27,9 @@ const Container = glamorous.view({
   borderColor: c.iosLightBackground,
 })
 
+const GITHUB_URL =
```
Ditto to my comment about adding a central constant for this.  Also, does the body of this const need to be on a different line?

hawkrives replied:
Say "hi" to prettier – the string + assignment is over 80 chars long, so prettier did what it does.

rye commented on source/lib/report-network-problem.js:
```diff
@@ -0,0 +1,10 @@
+import {tracker} from '../analytics'
+import bugsnag from '../bugsnag'
+
+export function reportNetworkProblem(err: Error) {
```
This is a good thing to centralize.

rye reviewed (changes requested):
Just a couple of off-hand code style tweaks is all I can offer at this point, but it looks good to me otherwise.

hawkrives committed:
move BusView's custom date into initialState

hawkrives committed:
remove props spread in BusStopRow

hawkrives committed:
fix time delay calculation in BusView

hawkrives committed:
use a const

hawkrives committed:
calculate a bunch of BusLine stuff once

and then store it in state, instead of calculating it on every render.

This also only re-renders the list when the "now" minute changes, to
further optimize it.

hawkrives committed:
bind setStateFromProps method

hawkrives committed:
clean up a setState call in BusView

hawkrives committed:
fix shouldComponentUpdate check in BusLine

It now only recalculates stuff when the time has changed, and it will
only re-render if the current list of bus stops/times has changed.
